eth/db/account.py

```diff
@@ -393,30 +359,42 @@ def _set_account(self, address: Address, account: Account) -> None:
     #
     # Record and discard API
     #
-    def record(self) -> Tuple[UUID, UUID]:
-        return (self._journaldb.record(), self._journaltrie.record())
-
-    def discard(self, changeset: Tuple[UUID, UUID]) -> None:
-        db_changeset, trie_changeset = changeset
-        self._journaldb.discard(db_changeset)
-        self._journaltrie.discard(trie_changeset)
+    def record(self) -> UUID:
+        changeset_id = self._journaldb.record()
+        self._journaltrie.record(changeset_id)
+        return changeset_id
+
+    def discard(self, changeset: UUID) -> None:
+        self._journaldb.discard(changeset)
+        self._journaltrie.discard(changeset)
         self._account_cache.clear()
 
-    def commit(self, changeset: Tuple[UUID, UUID]) -> None:
-        db_changeset, trie_changeset = changeset
-        self._journaldb.commit(db_changeset)
-        self._journaltrie.commit(trie_changeset)
+    def commit(self, changeset: UUID) -> None:
+        self._journaldb.commit(changeset)
+        self._journaltrie.commit(changeset)
 
     def make_state_root(self) -> Hash32:
-        self.logger.debug2("Generating AccountDB trie")
         self._journaldb.persist()
         self._journaltrie.persist()
         return self.state_root
 
     def persist(self) -> None:
-        self.make_state_root()
-        self._batchtrie.commit(apply_deletes=False)
-        self._batchdb.commit(apply_deletes=True)
+        self._validate_generated_root()
+        with self._raw_store_db.atomic_batch() as write_batch:
+            self._batchtrie.commit_to(write_batch, apply_deletes=False)
+            self._batchdb.commit_to(write_batch, apply_deletes=False)
```

Comment:
I know I'm picking on a lot of decisions which came before your PRs but this is my first time really interacting with these files :sweat_smile: I don't think `_batchtrie` and `_batchdb` need to be kept separate, since `commit_to` is always called at the same time changes to the trie and db are always tracked together.